### PR TITLE
Feat: add `continueFromHeader` methods to Sentry.

### LINKED ISSUE #1346: Missing continueFromHeaders method
When one wanna continue a trace from a header, right now it requires a bit of boilerplate and also no official docs on how to do it.

The guideline says we should provide a method called `continueFromHeaders` to make it easier https://develop.sentry.dev/sdk/performance/

eg Python https://github.com/getsentry/sentry-python/blob/2df9e1a230f1294b4fc319cb65838dcd6bb2e75c/sentry_sdk/tracing.py#L272

related to https://github.com/getsentry/sentry-docs/issues/2984

### PR DESCRIPTION
**Pull request by maciejwalkowiak**
## :scroll: Description

Add `continueFromHeader` methods to Sentry class.

## :bulb: Motivation and Context

Fixes #1346

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I added tests to verify the changes
- [ ] I updated the docs if needed
- [x] No breaking changes

### CODE REVIEW

File: sentry/src/main/java/io/sentry/Sentry.java [hunk 1]
```diff
@@ -667,6 +668,42 @@ public static void endSession() {
...
+  /**
+   * Starts a Transaction being a continuation of the span described in a `header` parameter.
+   * Started transaction is set on the scope.
+   *
+   * @param name the transaction name
+   * @param operation the operation
+   * @param header the string representing {@link SentryTraceHeader}
+   * @return created transaction
+   */
+  public static ISpan continueFromHeader(
```
Comment (marandaneto) [inline sentry/src/main/java/io/sentry/Sentry.java:680]:
```suggestion
  public static @NotNull ISpan continueFromHeader(
```

File: sentry/src/main/java/io/sentry/Sentry.java [hunk 2]
```diff
@@ -667,6 +668,42 @@ public static void endSession() {
...
+  /**
+   * Starts a Transaction being a continuation of the span described in a `header` parameter.
+   *
+   * @param name the transaction name
+   * @param operation the operation
+   * @param header the string representing {@link SentryTraceHeader}
+   * @param bindToScope if transaction should be bound to scope
+   * @return created transaction
+   */
+  public static ISpan continueFromHeader(
```
Comment (marandaneto) [inline sentry/src/main/java/io/sentry/Sentry.java:696]:
```suggestion
  public static @NotNull ISpan continueFromHeader(
```

File: sentry/src/main/java/io/sentry/Sentry.java [hunk 3]
```diff
@@ -667,6 +668,42 @@ public static void endSession() {
...
+   * Started transaction is set on the scope.
+   *
+   * @param name the transaction name
+   * @param operation the operation
+   * @param header the string representing {@link SentryTraceHeader}
+   * @return created transaction
+   */
+  public static ISpan continueFromHeader(
+      final @NotNull String name, final @NotNull String operation, final @NotNull String header)
+      throws InvalidSentryTraceHeaderException {
```
Comment (marandaneto) [inline sentry/src/main/java/io/sentry/Sentry.java:682]: not sure if I like the idea of `InvalidSentryTraceHeaderException`, so every call to this method would need to be wrapped within a try catch, the other option would be to swallow it and return a NoOpTransaction

wdyt @bruno-garcia ?

Comment (maciejwalkowiak) [reply sentry/src/main/java/io/sentry/Sentry.java:682]: Or start a new transaction without a parent context

Comment (bruno-garcia) [reply sentry/src/main/java/io/sentry/Sentry.java:682]: Agree it isn't great to throw. 
There's not much the user can do when they catch that other than "log with Sentry?"

Returning a new trace without a parent is likely better but best would be to log an error so when the time comes to debug this, we can logs. Log could print out the broken header too.

**Review by bruno-garcia — COMMENTED** [review]
I'm honestly not a fan of adding this at all.

It's a new public API in the static Sentry class for something trivial like we can achieve already with:

`startTransaction(
        TransactionContext.fromSentryTrace(name, operation, new SentryTraceHeader(header)));`

If it was for me we wouldn't merge this.

File: sentry/src/main/java/io/sentry/Sentry.java (hunk 3 above)
Comment (marandaneto) [reply sentry/src/main/java/io/sentry/Sentry.java:682]: wfm, not sure if useful but the new transaction could have a tag or extra or whatsoever with the invalid header so they know what happened if its not linked at all

**Comment by maciejwalkowiak** [comment]
@bruno-garcia @marandaneto lets discuss it on Friday then. I would also like to avoid adding this method.

**Comment by marandaneto** [comment]
> @bruno-garcia @marandaneto lets discuss it on Friday then. I would also like to avoid adding this method.

lets do the docs for now then please.

See https://github.com/getsentry/sentry-docs/issues/2984
there's no documented way on how to do it manually, in both cases

**Comment by bruno-garcia** [comment]
> > @bruno-garcia @marandaneto lets discuss it on Friday then. I would also like to avoid adding this method.
> 
> lets do the docs for now then please.
> 
> See [getsentry/sentry-docs#2984](https://github.com/getsentry/sentry-docs/issues/2984)
> there's no documented way on how to do it manually, in both cases

This sounds like a good approach for now.
Specially after our last call, it seems the best we can do now is stay true to the current spec, document what we have and start working on a bigger leap forward in terms of API. Meaning avoiding adding any extensions to the protocol to resolve use cases as we find them out.

**Comment by maciejwalkowiak** [comment]
Story continues in https://github.com/getsentry/sentry-docs/pull/3324

**Pull request closed without merging**